### PR TITLE
add missing libraries and correct bug

### PR DESCRIPTION
**Pull request by ew367**
# Description

This pull request will update the installLibraries.r script to add in missing packages needed to run QC.rmd and fix bugs to specify packager installer for V2 arrays and to install remotes:: package to allow install from github repos

## Issue ticket number

This pull request is to address issue: #221

## Type of pull request

- [x] Bug fix
- [ ] New feature/enhancement
- [ ] Code refactor
- [ ] Documentation update

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my code to check that it is functional
- [ ] I have used linters to check for common sources of errors
- [ ] I have implemented fail safes in my code to account for edge cases
- [ ] I have made the corresponding changes to the documentation

### CODE REVIEW

**Review by sof202 — CHANGES_REQUESTED** [review]

File: array/DNAm/preprocessing/installLibraries.r [hunk 1]
```diff
@@ -1,104 +1,105 @@
...
+  message("matrixStats version is > 1.1.0, downgrading to version 1.1.0")
+  remotes::install_version("matrixStats", version = "1.1.0", quiet = TRUE, repos = "https://cran.r-project.org")
+}
+packageVersion("matrixStats")
+
+#---------------------------------------------------------------------#
+# INSTALL PACKAGES FOR BRAIN CELL PROPORTION PREDICTION
+#---------------------------------------------------------------------#
+#Creating QC reports
+pkgs_rep <- c("pander","kableExtra", "gplots", "diptest", "corrplot", "pheatmap", "mixtools")
```
[the remark this answers is not in the document]
Comment (sof202) [reply array/DNAm/preprocessing/installLibraries.r:71]: We also use `reshape2` in the CETYGOdeconvolution.r script

[the remark this answers is not in the document]
Comment (sof202) [reply array/DNAm/preprocessing/installLibraries.r:71]: We are also relying on the `rmarkdown` library to generate the QC.html report.

https://github.com/ejh243/BrainFANS/blob/8e6dee660c58ac7bacabc7d97af48fcbb2325fd0/array/DNAm/preprocessing/jobSubmission/1_runDNAmQC.sh#L57

[the remark this answers is not in the document]
Comment (sof202) [reply array/DNAm/preprocessing/installLibraries.r:71]: And add `knitr` to the list as well (though `knitr` might just come with `rmarkdown` or `pander` already)

[the remark this answers is not in the document]
Comment (sof202) [reply array/DNAm/preprocessing/installLibraries.r:71]: Installing `rmarkdown` will also cover `knitr` as `knitr` is a dependency of the package

File: array/DNAm/preprocessing/installLibraries.r [hunk 2]
```diff
@@ -1,104 +1,105 @@
...
+  #library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
+  BiocManager::install("IlluminaHumanMethylation450kanno.ilmn12.hg19")
+  BiocManager::install("IlluminaHumanMethylation450kmanifest")
+}
+if(arrayType=='V1'){
+  BiocManager::install("IlluminaHumanMethylationEPICanno.ilm10b2.hg19")
+  BiocManager::install("IlluminaHumanMethylationEPICmanifest")
+}
+if(arrayType=='V2'){
+	BiocManager::install("jokergoo/IlluminaHumanMethylationEPICv2manifest")
```
[the remark this answers is not in the document]
Comment (sof202) [reply array/DNAm/preprocessing/installLibraries.r:49]: I had a look through the commit history and `library(BiocManager)` has never been used in this file. The lines must have just been copied and pasted in.

**Comment by ew367** [comment]
> Also, I presume that this installation script is purely for the 'preprocessing' part of the DNAm pipeline?

Yup

**Comment by ew367** [comment]
> You have CRLF (windows line endings) in the new version of the file. Could you change this back to LF (unix line endings)?

done

**Comment by ew367** [comment]
> You have CRLF (windows line endings) in the new version of the file. Could you change this back to LF (unix line endings)?

Done

File: array/DNAm/preprocessing/installLibraries.r (hunk 1 above)
[the remark this answers is not in the document]
Comment (ew367) [reply array/DNAm/preprocessing/installLibraries.r]: Good point. I've added these in

**Review by sof202 — APPROVED** [review]
Cool. We could also use `setdiff` with the `BiocManager::install` calls, but I'll leave that for a different PR.

**Pull request merged**